wrapper/src/test/java/integration/container/aurora/postgres/AuroraPostgresReadWriteSplittingPerformanceTest.java

```diff
@@ -0,0 +1,354 @@
+    setReadOnlyPerfDataList.add(dataTrue);
+
+    final PerfStatSetReadOnly dataFalse = new PerfStatSetReadOnly();
+    dataFalse.setReadOnly = "False";
+    dataFalse.minOverheadTime = TimeUnit.NANOSECONDS.toMillis(setReadOnlyFalseMinOverhead);
+    dataFalse.maxOverheadTime = TimeUnit.NANOSECONDS.toMillis(setReadOnlyFalseMaxOverhead);
+    dataFalse.avgOverheadTime = TimeUnit.NANOSECONDS.toMillis(setReadOnlyFalseAvgOverhead);
+    setReadOnlyPerfDataList.add(dataFalse);
+  }
+
+  @ParameterizedTest
+  @MethodSource("executeStatementsParameters")
+  public void test_readerLoadBalancing_executeStatements(final Properties props)
+      throws SQLException {
+    // This test isolates how much overhead is caused by reader load-balancing.
+    Long readerSwitchExecuteStatementsStartTime;
+    final List<Long> elapsedReaderSwitchExecuteStatementsTimes = new ArrayList<>(REPEAT_TIMES);
+    final PerfStatExecuteQueries data = new PerfStatExecuteQueries();
+
+    final String plugins = props.getProperty(PropertyDefinition.PLUGINS.name);
+
+    if (plugins != null && plugins.contains("readWriteSplitting")) {
+      data.pluginEnabled = "Enabled";
+      props.setProperty(ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.name, "true");
+    } else {
+      data.pluginEnabled = "Disabled";
+    }
+
+    for (int i = 0; i < REPEAT_TIMES; i++) {
+      try (final Connection conn = connectToInstance(
+          POSTGRES_CLUSTER_URL,
+          AURORA_POSTGRES_PORT,
+          props)) {
+
+        conn.setReadOnly(true);
+        try (final Statement stmt1 = conn.createStatement()) {
+          stmt1.executeQuery(QUERY_FOR_INSTANCE);
+        }
+
+        // The plugin does not switch readers on the first execute, so we'll start the timer after
+        for (int j = 0; j < EXECUTE_QUERY_TIMES; j++) {
+          // timer start
+          readerSwitchExecuteStatementsStartTime = System.nanoTime();
+          try (Statement stmt2 = conn.createStatement()) {
+
+            // timer end
+            final long readerSwitchExecuteStatementsTime =
+                (System.nanoTime() - readerSwitchExecuteStatementsStartTime);
+            elapsedReaderSwitchExecuteStatementsTimes.add(readerSwitchExecuteStatementsTime);
+
+            stmt2.executeQuery(QUERY_FOR_INSTANCE);
+          }
+        }
+      }
+    }
+
+    final long minAverageExecuteStatementTime =
+        elapsedReaderSwitchExecuteStatementsTimes.stream().min(Long::compare).orElse(0L);
+    final long maxAverageReaderSwitchExecuteStatementTime =
+        elapsedReaderSwitchExecuteStatementsTimes.stream().max(Long::compare).orElse(0L);
+    final long avgAverageExecuteStatementTime =
+        (long) elapsedReaderSwitchExecuteStatementsTimes.stream().mapToLong(a -> a).summaryStatistics().getAverage();
+
+    data.minAverageExecuteStatementTime =
+        round((minAverageExecuteStatementTime / (double) TimeUnit.MILLISECONDS.toNanos(1)), 3);
```

Comment:
You can store `(double) TimeUnit.MILLISECONDS.toNanos(1)` to a constant to reduce code duplication.